packages/lib-runtime/src/store/PluginStore.ts

```diff
@@ -25,6 +25,8 @@ type NotLoadedPluginInfo = {
   status: 'pending' | 'failed';
 };
 
+export type PluginInfo = LoadedPluginInfo | NotLoadedPluginInfo;
```

Comment:
Do we want to enforce define before use?  https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-use-before-define.md

Reply:
You can inspect the actual ESLint config for this file:
```sh
cd packages/lib-runtime
yarn lint:print-config src/store/PluginStore.ts
```

The relevant bits:
```jsonc
"no-use-before-define": [
  "off",
  {
    "functions": true,
    "classes": true,
    "variables": true
  }
],
"@typescript-eslint/no-use-before-define": [
  "error"
],
```

[`@typescript-eslint/no-use-before-define`](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-use-before-define.md) rule is active, using its default configuration.

By default, it doesn't care about type-related symbols, but it _does_ care about JS symbols, for example:
```ts
const a = x(); // triggers error: @typescript-eslint/no-use-before-define
const x = () => true;
```

Personally, I'm in favor of not enforcing this for type-related symbols.

Reply:
fair. Just noticed it.